technologies/job/nifi/1.9.2/instanceActions.js

```diff
@@ -0,0 +1,106 @@
+  } catch (error) {
+    return Response.error(`Fail to job "${job.featuresValues.processgroups.name}"`, { error, "url": `${job.featuresValues.endpoint.url}/nifi-api/process-groups/${formParams.processgroups.id}` });
+  }
+};
+
+/**
+ * Logic to stop the external job instance.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.stop = async ({ job, instance }) => {
+  try {
+    await axios.put(
+      `${job.featuresValues.endpoint.url}/nifi-api/flow/process-groups/${job.featuresValues.processgroups.id}`,
+       ({
+         "id": job.featuresValues.processgroups.id,
+         "state":"STOPPED",
+         "disconnectedNodeAcknowledged":false
+      }),
+    );
+
+    return Response.success();
+  } catch (error) {
+    return Response.error(`Fail to job "${job.featuresValues.processgroups.name}"`, { error, url: `${job.featuresValues.endpoint.url}/nifi-api/process-groups/${formParams.processgroups.id}` });
+  }
+};
+
+/**
+ * Logic to retrieve the external job instance status.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getStatus = async ({ job, instance }) => {
+  try {
+    console.log('GET LOG INSTANCE:', instance);
+    const { data } = await axios.get(
+      `${job.featuresValues.endpoint.url}/nifi-api/flow/process-groups/${job.featuresValues.processgroups.id}/status`,
+    );
+    switch (data.processGroupStatus.aggregateSnapshot.processorStatusSnapshots[0].processorStatusSnapshot.runStatus) {
+      case 'Running':
+        return Response.success(JobStatus.RUNNING);
+      case 'Stopped':
+        return Response.success(JobStatus.KILLED);
+      default:
+        return Response.success(JobStatus.AWAITING);
+    }
+  } catch (error) {
+    return Response.error(`Failed to get status for job ${job.featuresValues.processgroups.id}`, { error });
+  }
+
+};
+
+/**
+ * Logic to retrieve the external job instance logs.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getLogs = async ({ job, instance }) => {
+  try {
+    console.log('GET LOG INSTANCE:', instance);
+    const { data: logs } = await axios.get(
+      `${job.featuresValues.endpoint.url}/nifi-api/flow/bulletin-board?groupId${job.featuresValues.processgroups.id}`,
+    );
+    const output = logs.bulletinBoard.bulletins
+      .map(item => {
+        const d=new Date();
+        const isodate = (new Date(d.getUTCMonth()+ ' ' + d.getUTCDate() + ', ' + d.getUTCFullYear() + ' '+ item.bulletin.timestamp)).toISOString();
```

Comment:
Just to check, what is the format of the `timestamp` attribute on `bulletin` ?

I checked the NiFi API Documentation, but no example is provided, I just got the following information:

```
timestamp | string | optional | When this bulletin was generated.
```

I'm asking this because I think that this line can be simpler.

Reply:
You don't want to know, really.

It looks like this : '05:55:59 UTC' (yes no days, no month, no year)

Reply:
Oh, fine, I know understand why it's not very well documented 😂 
Thanks a lot. We'll keep the line as it is 👍